carmin-server/server/resources/authenticate.py

```diff
@@ -7,24 +7,25 @@
 from .models.authentication_credentials import AuthenticationCredentials, AuthenticationCredentialsSchema
 from server.common.error_codes_and_messages import INVALID_USERNAME_OR_PASSWORD
 from server.database.models.user import User
-from .decorators import marshal_request, marshal_response, login_required
+from .decorators import unmarshal_request, marshal_response, get_db_session
 from server.common.util import generate_api_key
 
 
 class Authenticate(Resource):
-    @marshal_request(AuthenticationCredentialsSchema())
+    @get_db_session
```

Comment:
I actually believe that this should be a decorator. It is a simple way for our endpoints that do require a safe database session to get one. Plus, when you look at the function, you can, in a glance, see that the function requires: an unmarshalled model, interracts with the database and is reserved to admins only.

Reply:
Discussed and agreed to keep it for the moment. In case we feel like needing extra decorators in the future, we will reconsider its importance further and possibly move it to a utils fonction.